core/shared/src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -53,6 +77,41 @@ object Argument extends PlatformArguments {
     override def defaultMetavar: String = typeName
   }
 
+  implicit val declineArgumentFunctor: Functor[Argument] =
+    new Functor[Argument] {
+      def map[A, B](arga: Argument[A])(fn: A => B): Argument[B] =
+        arga.map(fn)
+    }
+
+  implicit val declineArgumentSemigroupK: SemigroupK[Argument] =
+    new SemigroupK[Argument] {
+      override def combineK[A](x: Argument[A], y: Argument[A]): Argument[A] =
+        new Argument[A] {
+          override def read(string: String): ValidatedNel[String, A] = {
+            val ax = x.read(string)
+            if (ax.isValid) ax
+            else y.read(string)
+          }
+
+          override def defaultMetavar: String = s"${x.defaultMetavar}-or-${y.defaultMetavar}"
```

Comment:
Thoughts on ` | ` instead of `-or-`? Usage will read like `--my-option <string | url>` vs. `--my-option <string-or-url>`.

Interested because pipe is already used in similar contexts in usage to express alternatives, it may read better if you end up with `a | b | c | d`, and users sometimes underscore their metavars which would make `-or-` a style mismatch.